cypress/component/annot-categorical.cy.js

```diff
@@ -121,4 +126,15 @@ describe("Categorical annotation", () => {
         // Assert
         cy.get("@commitSpy").should("have.been.calledOnceWith", "changeMissingStatus", "column1", "HC", true);
     });
+
+    it("Can deal with an empty options array without crashing", () => {
+
+        // Setup
+        cy.mount(annotCategorical, {
+            computed: Object.assign(store.getters, { getCategoricalOptions: () => (p_column) => [],
```

Comment:
> the reasoning behind this Object.assign

this just creates a new Object of `store.getters` where the `getCategoricalOptions` and `getSelectedOptions` attributes are overwritten with what's shown above without changing the original `store` object defined at the beginning. An alternative way to do this would be to add the `beforeEach` hook to define the `store` and then directly change the `store` object in the setup of this test `store.getters.getCategoricalOptions = ...`. The main goal here is to change how these getters behave for this specific test from how they are defined at the beginning.

> the null return value of getSelectedOption mock

as mentioned in the test name, the goal here is to just check whether the component is able to deal with a `null` return value from the getter. The reason for this is that we now expect the getter to return an object where it previously returned a string literal - and I expect the (yet to be implemented) getter to return `null` if no selection has been made so far. So we want to be sure that the component can handle that.

Reply:
>this just creates a new Object of store.getters where the getCategoricalOptions and getSelectedOptions attributes >are overwritten with what's shown above without changing the original store object defined at the beginning. An >alternative way to do this would be to add the beforeEach hook to define the store and then directly change the store >object in the setup of this test store.getters.getCategoricalOptions = .... The main goal here is to change how these >getters behave for this specific test from how they are defined at the beginning.

Okay for now. I'll convert this to the `beforeEach` when I merge these changes with my PR changes. One of the reasons I asked about the `Object.assign` is I've seen some oddness with `store.getters` functions being shown as undefined by Cypress. The solution was to redefine the `store` in a `beforeEach` before each test.

>as mentioned in the test name, the goal here is to just check whether the component is able to deal with a null return >value from the getter. The reason for this is that we now expect the getter to return an object where it previously >returned a string literal - and I expect the (yet to be implemented) getter to return null if no selection has been made >so far. So we want to be sure that the component can handle that.

👍